dags/mozaggregator_prerelease.py

```diff
@@ -44,3 +54,51 @@
     ),
     dag=dag,
 )
+
+task_id = "prerelease_telemetry_aggregate_view_dataproc"
+gcp_conn = GoogleCloudBaseHook("google_cloud_airflow_dataproc")
+keyfile = json.loads(gcp_conn.extras["extra__google_cloud_platform__keyfile_dict"])
+client_email = keyfile["client_email"]
+
+prerelease_telemetry_aggregate_view_dataproc = SubDagOperator(
+    task_id=task_id,
+    dag=dag,
+    subdag=moz_dataproc_pyspark_runner(
+        parent_dag_name=dag.dag_id,
+        dag_name=task_id,
+        job_name="prerelease aggregates",
+        cluster_name="prerelease-telemetry-aggregates",
+        num_workers=10,
+        default_args=default_args,
+        init_actions_uris=[
+            "gs://dataproc-initialization-actions/python/pip-install.sh"
+        ],
+        additional_properties={
+            "spark:spark.jars": "gs://spark-lib/bigquery/spark-bigquery-latest.jar"
+        },
+        additional_metadata={
+            "PIP_PACKAGES": "git+https://github.com/mozilla/python_mozaggregator.git"
+        },
+        python_driver_code="gs://moz-fx-data-prod-airflow-dataproc-artifacts/jobs/mozaggregator_runner.py",
+        py_args=[
+            "aggregator",
+            "--date",
+            "{{ ds_nodash }}",
+            "--channels",
+            "nightly,aurora,beta",
+            "--credentials-bucket",
+            "telemetry-spark-emr-2",
+            "--credentials-prefix",
+            "aggregator_dev_database_envvars.json",
+            "--num-partitions",
+            10 * 32,
+            "--source",
+            "bigquery",
+            "--project-id",
+            "moz-fx-data-shared-prod",
+        ],
+        aws_conn_id="aws_dev_iam_s3",
+        gcp_conn_id=gcp_conn.gcp_conn_id,
+        service_account=client_email,
```

Comment:
Why do we need `service_account` here?

I'm asking because I haven't been providing it in update_orphaning and hardware report jobs. cc @haroldwoo

Reply:
The gcp_conn_id is used by airflow for permissions to create/delete dataproc clusters, and run dataproc workloads that only touch services in the same airflow gcp project. The cluster is created with the project's default service account, which is different from the gcp_conn_id's service account, unless we specify service_account here.

Specifying a service account here instantiates the dataproc cluster with a custom service account, and we usually do this when we need cross project permissions. Its just confusing because we often re-use the same service account for the gcp_conn_id and service_account params here.

Reply:
I'm putting this here in particular because I'm trying to test locally. I've updated the `google_cloud_airflow_dataproc` connection to insert my keyfile for `amiyaguchi-dev`. This will default to the service account associated with the production account otherwise.

```bash
function format_extras() {
KEYFILE="$1" python3 - <<END
import os
import json

with open(os.environ["KEYFILE"]) as f:
    data = json.load(f)

extra = {
    "extra__google_cloud_platform__project": data["project_id"],
    "extra__google_cloud_platform__keyfile_dict": json.dumps(data)
}

print(json.dumps(extra))
END
}

function update() {
    local conn_id=$1
    local keyfile=$2

    docker exec $container_id \
        airflow connections -d --conn_id $conn_id
    
    docker exec $container_id \
        airflow connections -a \
            --conn_id $conn_id \
            --conn_type google_cloud_platform \
            --conn_extra "$(format_extras $keyfile)"
}

container_id=$(docker ps | grep web | cut -d' ' -f1)
update google_cloud_airflow_dataproc <path to keyfile>
```